src/shared/components/layouts/SecondaryItems.tsx

```diff
@@ -0,0 +1,120 @@
+import { CSSProperties, useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import {
+  HelpIcon,
+  EnvelopeIcon,
+  BasketIcon,
+  ToolboxIcon,
+  Bubble,
+} from 'franklin-sites';
+import colors from '../../../../node_modules/franklin-sites/src/styles/colours.json';
+
+import useBasket from '../../hooks/useBasket';
+
+import { pluralise } from '../../utils/utils';
+
+import { LocationToPath, Location } from '../../../app/config/urls';
+import { RootState } from '../../../app/state/rootInitialState';
+
+import styles from './styles/secondary-items.module.scss';
+import { Status } from '../../../tools/types/toolsStatuses';
+
+interface Style extends CSSProperties {
+  // TODO: define and extend the supported custom properties in franklin
+  // TODO: find a way to expose them globally when using franklin elements
+  '--main-bubble-color': string;
+}
+
+const secondaryItemIconSize = '1.4em';
+
+const ToolsDashboard = () => {
+  const [count, failure] = useSelector<
+    RootState,
+    [count: number, failure: boolean]
+  >((state) => {
+    const unseenJobs = Object.values(state.tools).filter(
+      (job) => 'seen' in job && job.seen === false
+    );
+    return [
+      unseenJobs.length,
+      // at the moment, will always be false
+      unseenJobs.some((job) => job.status === Status.FAILURE),
+    ];
+  });
+
+  return (
+    <span title="Tools dashboard" className={styles['secondary-item']}>
+      <ToolboxIcon width={secondaryItemIconSize} />
+      {count ? (
+        <Bubble
+          className={styles.bubble}
+          style={
+            failure
+              ? ({ '--main-bubble-color': colors.failure } as Style)
+              : undefined
+          }
+          size="small"
+          title={`${count} new job ${pluralise('result', count)}`}
+        >
+          {count}
+        </Bubble>
+      ) : null}
+    </span>
+  );
+};
+
+export const Basket = () => {
+  const [basket] = useBasket();
+
+  const count = useMemo(
+    () =>
+      Array.from(basket.values())
+        .map((ns) => ns.size)
+        .reduce((total, current) => total + current, 0),
```

Comment:
My brain is not adept at understanding `reduce` so I tend to use lodash's `sum` in instances like this, but that's just personal preference

Reply:
Not a big fan of reduce myself, and doing sums is the only acceptable case for me. But that's a good suggestion and going with lodash I went a bit further and used `sumBy` that allowed me to drop also drop the `.map`